docs/tweetbox.md

```diff
@@ -0,0 +1,137 @@
+# Tweetbox
+
+In this section we'll implement a simple Tweetbox clone and learn how to assemble a view using custom tags.
+
+[Try it online](https://codepen.io/hyperapp/pen/bgWBdV?editors=0010)
+
+```jsx
+
+const MAX_LENGTH = 120
+const OFFSET = 10
+
+const OverflowText = ({ text, offset, count }) =>
+  <p>
+    ...{text.slice(0, offset)}
+    <span class="overflow-text">
+      {text.slice(count)}
+    </span>
+  </p>
+
+const Tweetbox = ({ count, text, update }) =>
+  <main>
+    <div class="container">
+      <ul class="flex-outer">
+        <li>
+          <textarea
+            placeholder="What's up?"
+            value={text}
+            oninput={update}
+          />
+        </li>
+
+        <li class="flex-inner">
+          <span class={count > OFFSET ? "" : "overflow-count"}>
+            {count}
+          </span>
+
+          <button
+            onclick={() => alert(text)}
+            disabled={count >= MAX_LENGTH || count <= 0}
+          >
+            Tweet
+          </button>
+        </li>
+      </ul>
+      {count < 0
+        ? <div class="overflow">
+            <h1>Whoops! Too long.</h1>
+            <OverflowText
+              text={text.slice(count - OFFSET)}
+              offset={OFFSET}
+              count={count}
+            />
+          </div>
+        : ""}
+
+    </div>
+  </main>
+
+app({
+  state: {
+    text: "",
+    count: MAX_LENGTH
+  },
+  view: (state, actions) =>
+    <Tweetbox
+      text={state.text}
+      count={state.count}
+      update={e => actions.update(e.target.value)}
+    />,
+  actions: {
+    update: (state, actions, text) => ({
+      text,
+      count: state.count + state.text.length - text.length
+    })
+  }
+})
+```
+
+The state stores the text of the message and the number of remaining characters `count`, initialized to `MAX_LENGTH`.
+
+```js
+state: {
+  text: "",
+  count: MAX_LENGTH
+}
+```
+
+The view consists of our custom TweetBox tag. We use the attributes, often referred to as _props_, to pass down data into the widget.
+
+```jsx
+<Tweetbox
+  text={state.text}
+  count={state.count}
+  update={e => actions.update(e.target.value)}
+/>
+```
+
+When the user types in the input, we call `actions.update` to update the current text and calculate the remaining characters.
+
+To calculate the number of remaining characters, we subtract the new text length from the current text length and add it to `state.count`.
```

Comment:
I find this just a tad convoluted and hard to follow. Probably just because I'm not thinking about the problem in the same way. Anyhow, I might have worded this something like:

The subtracting the length of the current text, from the length of the previous text, tells us how the number of remaining characters has changed. Hence the new count of remaining characters is the old count plus the aforementioned difference.